### PR TITLE
Integrate structure test framework

### PR DESCRIPTION
**Pull request by nkubala**
This code translates the existing structure tests from the nodejs-docker build into tests running on the GCP runtimes-common structure test framework. Necessary build artifacts for a configured container build are added and integrated into the preexisting jenkins build. The build will need to be changed to just the new script within the repository, which will take care of the rest.

dependent on https://github.com/GoogleCloudPlatform/runtimes-common/pull/49.

@dlorenc @JustinBeckwith

### CODE REVIEW

**Comment by nkubala** [comment]
@ofrobots @JustinBeckwith @dlorenc friendly ping

**Comment by ofrobots** [comment]
Where does the output for the tests go now?

**Comment by nkubala** [comment]
> Where does the output for the tests go now?

what I'm proposing would integrate these tests into the existing Jenkins run (by running the jenkins_run.sh script), which runs the tests and pipes all the output to stdout. since that would be running on the Jenkins host I'm assuming it would be captured in the build logs.

**Comment by dlorenc** [comment]
> what I'm proposing would integrate these tests into the existing Jenkins run (by running the jenkins_run.sh script), which runs the tests and pipes all the output to stdout. since that would be running on the Jenkins host I'm assuming it would be captured in the build logs.

Could you add the -v flag so we get more logging, including the test output?

**Comment by nkubala** [comment]
> Could you add the -v flag so we get more logging, including the test output?

it's on by default, as an argument to the structure test image in the cloubuild.yaml. https://github.com/nkubala/nodejs-docker/blob/595c0425a9acfaeb4db50efa6e6ad030a6cfb9d1/base/cloudbuild.yaml.in#L7

File: test/test.js [hunk 1]
```diff
@@ -25,24 +25,6 @@ describe('nodejs-docker', () => {
       }, 3000);
     });
   });
-
-  it('can use a custom nodejs version', (done) => {
```
Comment (ofrobots) [inline test/test.js]: This test used to get run on travis before, but now it doesn't. Conversely, the other tests in this file don't get run as part of the structure_tests. Is there a way to reconcile these?

File: test/test_config.yaml [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+
+- name: 'Custom Node Version'
+  setup: ['install_node v5.9.0']
+  teardown: ['install_node v6.9.1']
+  command: 'node'
+  flags: ['-v']
+  expectedOutput: ['v5.9.0\n']
+
+fileExistenceTests:
+- name: 'Semvar'
```
Comment (ofrobots) [inline test/test_config.yaml]: semver

File: test/test_config.yaml [hunk 3]
```diff
@@ -0,0 +1,30 @@
...
+  command: 'which'
+  flags: ['node']
+  expectedOutput: ['/nodejs/bin/node\n']
+
+- name: 'Node Version'
+  command: 'node'
+  flags: ['-v']
+  expectedOutput: ['v6.9.1\n']
+
+- name: 'Custom Node Version'
```
Comment (ofrobots) [inline test/test_config.yaml]: This test seems to be failing for me when I run build.sh manually.

Comment (nkubala) [inline test/test_config.yaml]: this test uses the setup and teardown fields, which are part of https://github.com/GoogleCloudPlatform/runtimes-common/pull/49. these tests will fail under this PR is merged and deployed

File: test/test_config.yaml (hunk 2 above)
Comment (nkubala) [inline test/test_config.yaml]: fixed

File: test/test.js (hunk 1 above)
Comment (ofrobots) [inline test/test.js]: In that case, can you keep this tests around. We do want travis to continue running this test as our CI for now.

Comment (nkubala) [inline test/test.js]: if you want to keep the tests in travis for now, I can add a script to the travis yaml to run the structure tests in addition to what you have running now. that way we can push forward on moving these tests to the framework/not have them duplicated, but you can keep them running through your normal CI.

Comment (ofrobots) [inline test/test.js]: SGTM.

**Comment by ofrobots** [comment]
Please do not merge changes without a LGTM! This change was probably good to merge, but it would be good to have the CI before merging.

**Comment by nkubala** [comment]
@ofrobots my apologies for jumping the gun here. I misinterpreted your SGTM above to be for the whole PR.

**Comment by ofrobots** [comment]
Retroactive LGTM based on a passing Travis build: https://travis-ci.org/GoogleCloudPlatform/nodejs-docker/builds/174939195.

**Pull request merged**